### PR TITLE
[develop] Trello-7rpc5tc9: Accounts

### PR DESCRIPTION
**Pull request by AndreySurzhan**
[Trello-7rpc5tc9]([https://trello.com/c/7rpc5tc9]): Accounts

- updated account controller
- updated project controller
- removed account routs
- accounts part of the projects
- soapUI auto-tests
screen shot 2017-10-24 at 11 23 20 pm

API
*PUT*
*GET*
`/projects/:id/accounts`
*PATCH*
*DELETE*
`/projects/:id/accounts/:accountId`

*Request*
```
{
	"name": "New Account - 1508901539142",
	"initValue" : 33333,
	"value": 66666,
	"currency": "578c6ef6832cff3b2cca13aa" 
}
```

*Response*
```
{
   "modifiedBy": "59f0029a0d4efb7111e18587",
   "createdBy": "59f0029a0d4efb7111e18587",
   "created": "2017-10-25T03:18:59.171Z",
   "currency":    {
      "_id": "578c6ef6832cff3b2cca13aa",
      "iso": "SEK",
      "created": "2016-07-18T05:53:58.714Z",
      "sign": "kr",
      "name": "Krona",
      "country": "Sweden",
      "modified": "2016-07-18T05:53:58.714Z"
   },
   "value": 66666,
   "initValue": 33333,
   "_id": "59f002a30d4efb7111e185ab",
   "modified": "2017-10-25T03:18:59.171Z",
   "name": "New Account - 1508901539142"
}
```

### CODE REVIEW

**Comment by AndreySurzhan** [comment]
@dimkahare @wildeastengineer if you like you can take a look. That is kind of final version.
All tests passed. But I've not review the code for typos or something.
Please let me know what you
Thanks

File: src/controllers/account.js [hunk 1]
```diff
@@ -1,165 +1,244 @@
...
+     */
+    updateAccount(data) {
+        let deferred = Q.defer();
+
+        ProjectModel.findOneAndUpdate({
+            _id: data.id,
+            owners: data.userId,
+            'accounts._id': data.account.id
+        }, {
+            $set: {
```
Comment (AndreySurzhan) [inline src/controllers/account.js]: We might have known issue with that one
https://trello.com/c/7goHFwf6
I am ok merging that PR without fixing that so far though

File: src/routes/project.js [hunk 2]
```diff
@@ -230,24 +255,101 @@ let ProjectRegisterRoutes = function (router, authenticate) {
...
+            userId: req.user._id,
+            account: {
+                id: req.params.accountId,
+                name: req.body.name,
+                initValue: req.body.initValue,
+                value: req.body.value,
+                currency: req.body.currency
+            }
+        })
+            .then((account) => {
```
Comment (wildeastengineer) [inline src/routes/project.js]: You can use here a construction like the next:

```
const accountParams = _.pick(req.body, 'name', 'initValue', 'value', 'currency');

projectController.updateAccount({
    id: req.params.id,
    userId: req.user._id,
    account: {
        id: req.params.accountId,
        ...accountParams
    }
})
```

If you like it

File: src/routes/project.js [hunk 3]
```diff
@@ -230,24 +255,101 @@ let ProjectRegisterRoutes = function (router, authenticate) {
...
+    /**
+     * Update given project account
+     *
+     * @function
+     * @name PATCH: /projects/:id/categories/:accountId
+     * @memberof routes/Project
+     *
+     * @returns {models/AccountSchema} - updated account
+     */
+    router.patch('/projects/:id/accounts/:accountId', authenticate, (req, res, next) => {
```
Comment (wildeastengineer) [inline src/routes/project.js]: Would it be better use `projectId` instead of `id` while we have `accountId` ?

File: src/routes/project.js (hunk 2 above)
Comment (AndreySurzhan) [reply src/routes/project.js]: had to update node to 8.8.1 from 7.5.3 to get it working

File: src/routes/project.js (hunk 3 above)
Comment (AndreySurzhan) [reply src/routes/project.js]: too many places to refactor.
is it ok with if we leave it as it for now?

**Pull request merged**